### PR TITLE
Add pyesptool as a programmer option

### PR DESCRIPTION
**Pull request by unreality**
I found that esptool consistently failed to upload on macos, this adds the python esptool as a programmer option

### CODE REVIEW

File: hardware/esp8266com/esp8266/platform.txt [hunk 1]
```diff
@@ -88,8 +88,14 @@ tools.esptool.cmd=esptool
...
-tools.esptool.upload.protocol=esp
-tools.esptool.upload.params.verbose=-vv
-tools.esptool.upload.params.quiet=
-tools.esptool.upload.pattern="{path}/{cmd}" {upload.verbose} -cd {upload.resetmethod} -cb {upload.speed} -cp "{serial.port}" -ca 0x00000 -cf "{build.path}/{build.project_name}_00000.bin" -ca 0x40000 -cf "{build.path}/{build.project_name}_40000.bin"
+tools.esptool.program.params.verbose=-vv
+tools.esptool.program.params.quiet= 
+tools.esptool.program.pattern="{path}/{cmd}" -vvv {program.verbose} -cd {upload.resetmethod} -cb {upload.speed} -cp {serial.port} -ca 0x00000 -cf "{build.path}/{build.project_name}_00000.bin" -ca 0x40000 -cf "{build.path}/{build.project_name}_40000.bin"
+
+tools.pyesptool.cmd=esptool.py
+tools.pyesptool.cmd.windows=esptool.exe
```
Comment (igrr) [inline hardware/esp8266com/esp8266/platform.txt]: windows version of command doesn't look right...

**Comment by unreality** [comment]
Latest commits should work for windows if they have python installed and available in %PATH%, but I have not been able to test it. Should help with #129 , #95 , #138 #3

**Comment by igrr** [comment]
I was looking at the option of packaging esptool.py along with python runtime using [pyinstaller](https://github.com/pyinstaller/pyinstaller). This would remove dependency on Python. Unfortunately, the binaries built using this method don't work on older versions of OS X (~10.6).

**Comment by igrr** [comment]
Esptool.py needs to be updated to work around a bug in the SPIEraseArea ROM function: https://github.com/themadinventor/esptool/issues/33.

**Comment by Links2004** [comment]
i build the fixed version for windows:
https://www.dropbox.com/s/jy9e25kcaceftjl/esptool.zip?dl=0

**Comment by circuitFlower** [comment]
just an fyi, I started fresh with a new esp8266, and got it up and running on my Ubuntu VM. I think the problem was that I was trying to use this pyesptool solution? At one point it said it was erasing the flash, went through to 100%, then hung. I never got that esp working again.

**Comment by rei-vilo** [comment]
From #78: On Mac OS X, esptools.py` requires `libusb` and works only with Python 2.7.

**Comment by zerog2k** [comment]
I'm having a possibly related problem. I'm on ubuntu and esptool-ck will fail to upload, whereas esptool.py will always work fine..

I'm setting up a custom board (10k pullups on ch_pd & reset, 10k pulldown on gpio0).. I hooked up logic analyser, there are some notable differences, but sequence of reset/rts and gpio0/dtr seem to be acceptable for both.

So assuming this is for adding esptool.py for flashing, then +1 for this effort.

**Pull request closed without merging**